pkg/common/vclib/datastore.go

```diff
@@ -170,3 +205,42 @@ func (dsi *DatastoreInfo) ListFirstClassDiskInfos(ctx context.Context) ([]*First
 
 	return objs, nil
 }
+
+// GetFirstClassDiskInfoByName gets a specific first class disks (FCD) on this datastore
+func (dsi *DatastoreInfo) GetFirstClassDiskInfoByName(ctx context.Context, diskName string) (*FirstClassDiskInfo, error) {
+	m := vslm.NewObjectManager(dsi.Datacenter.Client())
+
+	oids, err := m.List(ctx, dsi.Reference())
+	if err != nil {
+		glog.Errorf("Failed to list disks. Err: %v", err)
+		return nil, err
+	}
+
+	var ids []string
```

Comment:
Could drop this var and loop, just use `oids` as-is in the main loop.

Reply:
Can do... this is just a copy and paste from an existing function. Will get rid of the loop.